funnel/views/index.py

```diff
@@ -176,6 +177,60 @@ def opensearch():
     return render_template('opensearch.xml.jinja2')
 
 
+@app.route('/sitemap.xml')
+@cache.cached(timeout=7200)
+def sitemap():
+    sitemapxml = (
+        '<?xml version="1.0" encoding="UTF-8"?>\n'
+        '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">\n'
+    )
+    profiles = []
+    for profile in Profile.query.all():
+        if profile.state.PUBLIC:
+            profiles.append(profile.id)
+            sitemapxml += (
+                '  <url>\n'
+                '    <loc>{url}</loc>\n'
+                '    <lastmod>{updated_at}</lastmod>\n'
+                '    <changefreq>monthly</changefreq>\n'
+                '  </url>\n'.format(
+                    url=profile.url_for(_external=True),
+                    updated_at=profile.updated_at.isoformat(),
+                )
+            )
+    projects = []
+    for project in Project.query.all():
+        if project.state.PUBLISHED:
+            projects.append(project.id)
+            sitemapxml += (
+                '  <url>\n'
+                '    <loc>{url}</loc>\n'
+                '    <lastmod>{updated_at}</lastmod>\n'
+                '    <changefreq>monthly</changefreq>\n'
+                '  </url>\n'.format(
+                    url=project.url_for(_external=True),
+                    updated_at=project.updated_at.isoformat(),
+                )
+            )
+    proposals = []
+    for proposal in Proposal.query.all():
+        if not proposal.state.DELETED:
```

Comment:
Also have to exclude `DRAFT` proposals. Not sure about `CANCELLED` and `REJECTED`.